evap/static/ts/src/contact_modal.ts

```diff
@@ -0,0 +1,61 @@
+declare const bootstrap: typeof import("bootstrap");
+
+import { selectOrError, sleep, assert } from "./utils.js";
+import { csrfHeader } from "./csrf-utils.js";
+
+const success_message_timeout = 3000;
+
+export class ContactModalLogic {
+    private readonly modal: bootstrap.Modal;
+    private readonly successMessageModal: bootstrap.Modal;
+    private readonly actionButtonElement: HTMLButtonElement;
+    private readonly messageTextElement: HTMLInputElement;
+    private readonly anonymousRadioElement: HTMLInputElement;
+    private readonly modalId: string;
+    private readonly title: string;
+
+    constructor(modalId: string, title: string) {
+        this.modalId = modalId;
+        this.title = title;
+        this.modal = new bootstrap.Modal(selectOrError("#" + modalId));
+        this.successMessageModal = new bootstrap.Modal(selectOrError("#successMessageModal_" + modalId));
+        this.actionButtonElement = selectOrError("#" + modalId + "ActionButton");
+        this.messageTextElement = selectOrError("#" + modalId + "MessageText");
+        this.anonymousRadioElement = selectOrError("#" + modalId + "AnonymName");
+        this.actionButtonElement.addEventListener("click", async event => {
+            this.actionButtonElement.disabled = true;
+            event.preventDefault();
+            const message = this.messageTextElement.value;
+            if (message.trim() == "") {
+                this.modal.hide();
+                this.actionButtonElement.disabled = false;
+                return;
+            }
+            try {
+                const response = await fetch("/contact", {
+                    body: new URLSearchParams({
+                        anonymous: this.anonymousRadioElement.value,
```

Comment:
Wouldn't this always be "on", the default `value` attribute of the radio elements?